### PR TITLE
Feat/error pages

### LINKED ISSUE #65: [Feat]: Customize 404 and 500 error page
### My suggestion

So our error page looks like this and we don't like this, cfhange it! UwU
image

Here some reference
https://stackoverflow.com/questions/17662928/django-creating-a-custom-500-404-error-page
https://docs.djangoproject.com/en/5.0/ref/views/

### Possible alternatives

_No response_

### PR DESCRIPTION
**Pull request by hexxcatee**
I changed the default 404 and 500 error pages into custom ones with a simplistic and silly look :33

### CODE REVIEW

File: gwardia_hub/settings.py [hunk 1]
```diff
@@ -72,7 +74,7 @@
 TEMPLATES = [
     {
         "BACKEND": "django.template.backends.django.DjangoTemplates",
-        "DIRS": [],
+        "DIRS": [os.path.join(BASE_DIR)],
```
Comment (Norbiros) [inline gwardia_hub/settings.py]: Why? New error templates are in app (core) dir.

File: core/templates/core/500.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{% extends 'core/base.html' %}
+
+{% block content %}
+<div class="w-full h-80vh my-28 flex flex-col items-center justify-center space-y-8">
+    <div class="size-1/4">
+    <img src='https://cdn.discordapp.com/attachments/804080757127643166/1236116899935551549/NIGER.png?ex=6636d712&is=66358592&hm=f5c1b4fc0d3323602bd56e69f7ac2b3fc187d58f24d340bf54ceb540fad62688&'>
+    </div>
```
Comment (Norbiros) [inline core/templates/core/500.html]: Indent.
Put image to `static/`

File: core/templates/core/500.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    <div class="flex flex-col items-center justify-center">
+        <p class="text-7xl text-gray-600 font-dark font-extrabold mb-8">500</p>
+        <p class="text-2xl font-light leading-normal mb-8">Ojej, coś jest nie tak z naszym serwerem :(</p>
```
Comment (Norbiros) [inline core/templates/core/500.html]: You could add info, that you can report bug on github.

File: core/templates/core/404.html [hunk 4]
```diff
@@ -0,0 +1,15 @@
+{% extends 'core/base.html' %}
+
+{% block content %}
+<div class="h-80vh w-screen my-28 flex items-center">
```
Comment (Norbiros) [inline core/templates/core/404.html]: Here you use `w-screen` and in 500 you use `w-full`

File: core/templates/core/404.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    <div class="container flex md:flex-row items-center justify-between px-20 text-gray-700">
+        <div class="w-full mx-8">
+            <div class="text-7xl text-gray-600 font-dark font-extrabold mb-8">404</div>
+            <p class="text-2xl md:text-3xl font-light leading-normal mb-8">Ojej! Wygląda na to, że jesteś teraz na drodze do nikąd :o</p>
+        </div>
+        <div class="w-full lg:flex lg:justify-end lg:w-1/2 mx-5 my-12">
+            <img src="https://cdn.discordapp.com/attachments/1233197360465838161/1235649630729932942/uwu.png?ex=663523e5&is=6633d265&hm=25f895c70be1b5b5f92c171775ad9b4881c7d61d4b7e6e65d5a7188d1efa1603&"/>
```
Comment (Norbiros) [inline core/templates/core/404.html]: Put this image into `static/`

**Review by Norbiros — CHANGES_REQUESTED** [review]

File: core/templates/core/404.html (hunk 5 above)
Comment (Sanielia) [reply core/templates/core/404.html]: And add alt text

File: core/templates/core/404.html [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{% extends 'core/base.html' %}
+
+{% block content %}
+<div class="h-80vh w-screen my-28 flex items-center">
+    <div class="container flex md:flex-row items-center justify-between px-20 text-gray-700">
```
Comment (Sanielia) [inline core/templates/core/404.html]: Add color to each element, because now this bottom text is to dark 
image

```suggestion
    <div class="container flex md:flex-row items-center justify-between px-20">
```

File: core/templates/core/500.html (hunk 2 above)
Comment (Sanielia) [reply core/templates/core/500.html]: And add alt text

**Review by Sanielia — CHANGES_REQUESTED** [review]
This and what norbirosek asked to
<333

**Review by goteusz-maszyk — CHANGES_REQUESTED** [review]
Add 403 Forbidden too pls

**Review by Sanielia — CHANGES_REQUESTED** [review]
Fix unresolved chnages requested by @Norbiros and add 403 Forbidden page

**Review by Sanielia — DISMISSED** [review]
UwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwUUwU UwU UwU UwU UwU UwU UwU

**Review by Sanielia — DISMISSED** [review]
uwuwuwu

File: core/templates/core/error.html [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+{% block content %}
+<div class="h-80vh w-screen my-28 flex items-center">
+    <div class="container flex md:flex-row items-center justify-between px-20 text-gray-700">
+        <div class="w-full mx-8">
+            <div class="text-7xl text-gray-600 font-dark font-extrabold mb-8">{{ error_id }}</div>
+            <p class="text-2xl md:text-3xl font-light leading-normal mb-8">{{ description }}<br>Możesz zgłosić błąd <a class="text-indigo-700 font-bold" href='https://github.com/Gwardia-Czapli/GwardiaHub'>tutaj</a></p>
+        </div>
+        <div class="w-full lg:flex lg:justify-end lg:w-1/2 mx-5 my-12">
+            {% load static %}
+            <img src="{{ image }}"  alt="Error img"/>
```
Comment (Norbiros) [inline core/templates/core/error.html]:
```suggestion
            <img src="{{ image }}" alt="Error img"/>
```

File: core/templates/core/error.html [hunk 8]
```diff
@@ -0,0 +1,16 @@
+{% extends 'core/base.html' %}
+
+{% block content %}
+<div class="h-80vh w-screen my-28 flex items-center">
+    <div class="container flex md:flex-row items-center justify-between px-20 text-gray-700">
+        <div class="w-full mx-8">
+            <div class="text-7xl text-gray-600 font-dark font-extrabold mb-8">{{ error_id }}</div>
+            <p class="text-2xl md:text-3xl font-light leading-normal mb-8">{{ description }}<br>Możesz zgłosić błąd <a class="text-indigo-700 font-bold" href='https://github.com/Gwardia-Czapli/GwardiaHub'>tutaj</a></p>
```
Comment (Norbiros) [inline core/templates/core/error.html]: This line is too long. Move maybe `<a>` (maybe with `<br>`) to next line :3

File: core/views.py [hunk 9]
```diff
@@ -36,3 +37,30 @@ def gh_webhook(request):
...
+        "description": "Ojej, coś jest nie tak z naszym serwerem :(",
+        "image": static("images/500_image.png"),
+    }
+    return render(request, "core/error.html", context, status=500)
+
+
+def handling_403(request, exception):
+    context = {
+        "error_id": "403",
+        "description": "Brak dostępu :( ",
```
Comment (Norbiros) [inline core/views.py]: Maybe add information, that user can log in at `/login`

File: core/views.py [hunk 10]
```diff
@@ -36,3 +37,30 @@ def gh_webhook(request):
...
+def handling_500(request):
+    context = {
+        "error_id": "500",
+        "description": "Ojej, coś jest nie tak z naszym serwerem :(",
+        "image": static("images/500_image.png"),
+    }
+    return render(request, "core/error.html", context, status=500)
+
+
+def handling_403(request, exception):
```
Comment (Norbiros) [inline core/views.py]: Why sometimes functions have `request, exception`, but sometimes just `reqest`?

**Review by Norbiros — CHANGES_REQUESTED** [review]
:333

File: core/views.py (hunk 10 above)
Comment (hexxcatee) [reply core/views.py]: when i put only request in handling 403 there was an error. so should i just put both req and excp in all of the functions?

**Review by Sanielia — DISMISSED** [review]

File: core/templates/core/error.html [hunk 11]
```diff
@@ -0,0 +1,17 @@
+{% extends 'core/base.html' %}
+
+{% block content %}
+<div class="h-80vh w-screen my-28 flex items-center">
+    <div class="container flex md:flex-row items-center justify-between px-20 text-gray-700">
+        <div class="w-full mx-8">
+            <div class="text-7xl text-gray-600 font-dark font-extrabold mb-8">{{ error_id }}</div>
+            <p class="text-2xl md:text-3xl font-light leading-normal mb-8">{{ description }}
+                <br>Możesz zgłosić błąd <a class="text-indigo-700 font-bold" href='https://github.com/Gwardia-Czapli/GwardiaHub'>tutaj</a></p>
```
Comment (Norbiros) [inline core/templates/core/error.html]:
```suggestion
                <br>Możesz zgłosić błąd <a class="text-indigo-700 font-bold" href="https://github.com/Gwardia-Czapli/GwardiaHub">tutaj</a></p>
```

File: core/views.py [hunk 12]
```diff
@@ -1,26 +1,15 @@
...
+from django.shortcuts import render
+from django.templatetags.static import static
 
 
 def index(request):
-    context = {
-        "navbar_links": [
-            {
-                "name": "Panel",
-                "url": "panel/",
```
Comment (Norbiros) [inline core/views.py]: Unrelated random change, maybe mistake during rebase?

File: core/views.py [hunk 13]
```diff
@@ -29,10 +18,28 @@ def profile(request, name: str):
     return render(request, "core/profile.html", context={"user": user})
 
 
-@csrf_exempt
-@require_POST
-def gh_webhook(request):
```
Comment (Norbiros) [inline core/views.py]: <3

**Review by Norbiros — CHANGES_REQUESTED** [review]

**Review by Sanielia — DISMISSED** [review]

File: core/views.py [hunk 14]
```diff
@@ -17,10 +21,10 @@ def index(request):
     return render(request, "core/index.html", context)
 
 
-def profile(request, name: str):
+def profile(request):
     # Those are placeholder data
     user = {
-        "name": name,
+        "name": "Salieri",
```
Comment (Sanielia) [inline core/views.py]:
```suggestion
        "name": name,
```

File: core/views.py [hunk 15]
```diff
@@ -17,10 +21,10 @@ def index(request):
     return render(request, "core/index.html", context)
 
 
-def profile(request, name: str):
+def profile(request):
```
Comment (Sanielia) [inline core/views.py]:
```suggestion
def profile(request, name: str):
```

File: core/views.py [hunk 16]
```diff
@@ -36,3 +40,30 @@ def gh_webhook(request):
...
+def handling_500(request):
+    context = {
+        "error_id": "500",
+        "description": "Ojej, coś jest nie tak z naszym serwerem :(",
+        "image": static("images/500_image.png"),
+    }
+    return render(request, "core/error.html", context, status=500)
+
+
+def handling_403(request, exception):
```
Comment (Sanielia) [inline core/views.py]: Change order by moving 403 above 404

**Review by Sanielia — CHANGES_REQUESTED** [review]

**Review by Sanielia — DISMISSED** [review]

**Review by Norbiros — CHANGES_REQUESTED** [review]
Update branch hihi

File: core/templates/core/error.html [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+{% block content %}
+<div class="h-80vh w-screen my-28 flex items-center">
+    <div class="container flex md:flex-row items-center justify-between px-20 text-gray-700">
+        <div class="w-full mx-8">
+            <div class="text-7xl text-gray-400 font-dark font-extrabold mb-8">{{ error_id }}</div>
+            <p class="text-2xl md:text-3xl font-light leading-normal mb-8">{{ description|linebreaks }}
+                Możesz zgłosić błąd <a class="text-indigo-700 font-bold" href="https://github.com/Gwardia-Czapli/GwardiaHub">tutaj</a></p>
+        </div>
+        <div class="w-full lg:flex lg:justify-end lg:w-1/2 mx-5 my-12">
+            {% load static %}
```
Comment (Norbiros) [inline core/templates/core/error.html]: Is this line necessary?

**Review by Norbiros — CHANGES_REQUESTED** [review]

**Review by Sanielia — DISMISSED** [review]
<3

File: core/templates/core/error.html [hunk 18]
```diff
@@ -0,0 +1,16 @@
+{% extends 'core/base.html' %}
+
+{% block content %}
+<div class="h-screen w-screen my-28 flex items-center justify-center">
+    <div class="container flex md:flex-row items-center justify-between px-20 text-gray-700">
+        <div class="w-full mx-8">
+            <div class="text-8xl text-gray-400 font-dark font-extrabold mb-8">{{ error_id }}</div>
+            <p class="text-2xl font-light leading-normal mb-8">{{ description|linebreaksbr }}
+                <br>Możesz zgłosić błąd <a class="text-indigo-700 font-bold" href="https://github.com/Gwardia-Czapli/GwardiaHub">tutaj</a></p>
```
Comment (Norbiros) [inline core/templates/core/error.html]: Move `</p` to a new line.

**Review by Norbiros — CHANGES_REQUESTED** [review]
Almost lgtm

**Review by Norbiros — APPROVED** [review]

**Review by goteusz-maszyk — APPROVED** [review]
Looks ok
